src/lib/features/events/event-service.test.ts

```diff
@@ -0,0 +1,111 @@
+import type { ProjectAccess } from '../private-project/privateProjectStore';
+import { filterPrivateProjectsFromParams } from './event-service';
+
+describe('filterPrivateProjectsFromParams', () => {
+    it('should return IS_ANY_OF with allowed projects when projectParam is undefined and mode is limited', () => {
+        const projectAccess: ProjectAccess = {
+            mode: 'limited',
+            projects: ['project1', 'project2'],
+        };
+
+        const projectParam = undefined;
+
+        const result = filterPrivateProjectsFromParams(
+            projectParam,
+            projectAccess,
+        );
+
+        expect(result).toBe('IS_ANY_OF:project1,project2');
+    });
+
+    it('should return the original projectParam when mode is all', () => {
+        const projectAccess: ProjectAccess = {
+            mode: 'all',
+        };
+
+        const projectParam = 'IS:project1,IS:project3';
+
+        const result = filterPrivateProjectsFromParams(
+            projectParam,
+            projectAccess,
+        );
+
+        expect(result).toBe(projectParam);
+    });
+
+    it('should filter out projects not in allowedProjects when mode is limited', () => {
+        const projectAccess: ProjectAccess = {
+            mode: 'limited',
+            projects: ['project1', 'project2'],
+        };
+
+        const projectParam = 'IS:project1,IS:project3';
+
+        const result = filterPrivateProjectsFromParams(
+            projectParam,
+            projectAccess,
+        );
+
+        expect(result).toBe('IS:project1');
+    });
+
+    it('should return a single project if only one is allowed', () => {
+        const projectAccess: ProjectAccess = {
+            mode: 'limited',
+            projects: ['project1'],
+        };
+
+        const projectParam = 'IS:project1,IS:project2';
+
+        const result = filterPrivateProjectsFromParams(
+            projectParam,
+            projectAccess,
+        );
+
+        expect(result).toBe('IS:project1');
+    });
+
+    it('should throw an error if no projects match and not throw an error', () => {
```

Comment:
Huh? What's this test supposed to say?  Something like this?
```suggestion
    it('should throw an error if no projects match', () => {
```

Reply:
Yeah, fixed